code/_onclick/click_alt.dm

```diff
@@ -86,6 +85,48 @@
 	return NONE
 
 
+/**
+ * ### Base proc for alt click interaction right click.
+ *
+ * If you wish to add custom `click_alt_secondary` behavior for a single type, use that proc.
+ */
+/mob/proc/base_click_alt_secondary(atom/target)
+	SHOULD_NOT_OVERRIDE(TRUE)
+
+	if(SEND_SIGNAL(src, COMSIG_MOB_ALTCLICKON_SECONDARY, target) & COMSIG_MOB_CANCEL_CLICKON)
+		return
+
+	var/can_use_click_action = FALSE
+	if(isturf(target))
+		// Turfs are special because they can't be used with can_perform_action
+		can_use_click_action = can_perform_turf_action(target)
+	else
+		can_use_click_action = can_perform_action(target, target.interaction_flags_click | SILENT_ADJACENCY)
+	if(can_use_click_action)
+		target.alt_click_secondary(src)
+
+///The base proc of when something is right clicked on when alt is held
+/atom/proc/alt_click_secondary(mob/user)
```

Comment:
do we still need this?

Reply:
Yes. Alt click on the mob will call alt click on the atom which will then call alt_click_secondary()

Reply:
why not just get merge it with base_click_alt_secondary if that's the case?

Reply:
It has to do with SEND_SIGNAL. First we send it from the mob and then from the atom. If it wasn't for that them they could be merged

Reply:
I see what you mean. I don't really make the distinction in base click alt, so this seems just a stylistic difference

Reply:
You right. I'll remove one of the signals cause they are redundant